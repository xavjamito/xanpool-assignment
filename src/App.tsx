import React, { FC } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Layout } from "./components/Layout/Layout";
import Home from "./components/Home/Home";
import Repos from "./components/Repos/Repos";
import RepoInfo from "./components/RepoInfo/RepoInfo";
import "antd/dist/antd.css";


const App: FC = () => {
  return (
    <div className="App">
      <Router>
        <Layout>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/repos/:username" component={Repos} />
            <Route exact path="/repos/:username/:repoName" component={RepoInfo} />
          </Switch>
        </Layout>
      </Router>
    </div>
  );
}

export default App;
