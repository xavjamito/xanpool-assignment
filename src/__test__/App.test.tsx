import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

test('renders correctly', () => {
  const app = render(<App />);
  expect(app).toMatchSnapshot();
});
