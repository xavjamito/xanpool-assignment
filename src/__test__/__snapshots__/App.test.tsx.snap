// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="ant-page-header site-header ant-page-header-ghost"
        >
          <div
            class="ant-page-header-heading"
          >
            <div
              class="ant-page-header-heading-left"
            >
              <span
                class="ant-page-header-heading-title"
              >
                <a
                  href="/"
                >
                  Github User Repos
                </a>
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-bdnxRM gRWHsq"
        >
          <div
            class="sc-gtsrHT cZtdqt"
          >
            <h2>
               Github Repo Search 
            </h2>
            
            <span
              class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-search ant-input-search-large ant-input-search-with-button"
            >
              <span
                class="ant-input-wrapper ant-input-group"
              >
                <span
                  class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                >
                  <input
                    class="ant-input ant-input-lg"
                    placeholder="Search for Github username here"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                      role="button"
                      tabindex="-1"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <span
                  class="ant-input-group-addon"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
                    type="button"
                  >
                    <span>
                      Search
                    </span>
                  </button>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="ant-page-header site-header ant-page-header-ghost"
      >
        <div
          class="ant-page-header-heading"
        >
          <div
            class="ant-page-header-heading-left"
          >
            <span
              class="ant-page-header-heading-title"
            >
              <a
                href="/"
              >
                Github User Repos
              </a>
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-bdnxRM gRWHsq"
      >
        <div
          class="sc-gtsrHT cZtdqt"
        >
          <h2>
             Github Repo Search 
          </h2>
          
          <span
            class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-search ant-input-search-large ant-input-search-with-button"
          >
            <span
              class="ant-input-wrapper ant-input-group"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
              >
                <input
                  class="ant-input ant-input-lg"
                  placeholder="Search for Github username here"
                  type="text"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                    role="button"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
                  type="button"
                >
                  <span>
                    Search
                  </span>
                </button>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
